.github/workflows/bors.yml

```diff
@@ -154,14 +166,117 @@ jobs:
 
       - name: lint mathlib
+        id: lint
         uses: liskin/gh-problem-matcher-wrap@v2
         with:
           linters: gcc
           run: env LEAN_ABORT_ON_PANIC=1 make lint
 
+      - name: Post comments for lean-pr-testing branch
+        if: always()
+        env:
+          TOKEN: ${{ secrets.LEAN_PR_TESTING }}
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+          WORKFLOW_URL: ${{ github.event.workflow_run.html_url }}
+        run: |
+          # Extract branch name and check if it matches the pattern
+          branch_name=$(echo "$GITHUB_CONTEXT" | jq -r .ref | cut -d'/' -f3)
+          if [[ "$branch_name" =~ ^lean-pr-testing-([0-9]+)$ ]]; then
+            pr_number="${BASH_REMATCH[1]}"
+            current_time=$(date "+%Y-%m-%d %H:%M:%S")
+
+            echo "This is a 'lean-pr-testing-$pr_number' branch, so we need to adjust labels and write a comment."
+
+            if ${{ steps.lint.outcome == 'success' }}; then
+              echo "Removing label awaiting-mathlib-build"
+              curl -L -s \
+                -X DELETE \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels/awaiting-mathlib-build
+              echo "Removing label breaks-mathlib"
+              curl -L -s \
+                -X DELETE \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels/breaks-mathlib
+              echo "Adding label builds-mathlib"
+              curl -L -s \
+                -X POST \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels \
+                -d '{"labels":["builds-mathlib"]}'
+            else
+              echo "Removing label builds-mathlib"
+              curl -L -s \
+                -X DELETE \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels/builds-mathlib
+              echo "Adding label breaks-mathlib"
+              curl -L -s \
+                -X POST \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels \
+                -d '{"labels":["breaks-mathlib"]}'
+            fi
```

Comment:
I think this would be easier to follow and debug if written with https://github.com/actions/github-script, where adding a label is spelt
```js
octokit.rest.issues.addLabels({
  owner: "leanprover", repo: "lean4", issue_number: pr_number,
  labels: { name:"breaks-mathlib" }
})
```

Reply:
This would require a complete rewrite from bash to javascript, so I'm pretty unexcited about doing this right now.

I agree it would be better, but testing this stuff already cost me a large chunk of today. This code (mostly? hopefully?) works --- at least, I've exercised it.

Reply:
Per Mario's suggestion I have factored out the bash into an external script.